packages/aws-cdk/lib/util/please-hold.ts

```diff
@@ -0,0 +1,25 @@
+import { warning } from "../logging";
+
+/**
+ * Print a message to the logger in case the operation takes a long time
+ */
+export class PleaseHold {
+  private handle?: NodeJS.Timer;
+
+  constructor(private readonly message: string, private readonly timeoutSec = 10) {
+  }
+
+  public start() {
+    this.handle = setTimeout(this.printMessage.bind(this), this.timeoutSec * 1000);
+  }
+
+  public stop() {
+    if (this.handle) {
+      clearTimeout(this.handle);
+    }
+  }
+
+  private printMessage() {
+    warning(this.message);
```

Comment:
I don't reckon this deserves to be a warning. At best it's an information.

Reply:
Warning just applies nice coloring that I want.